packages/quick_actions/quick_actions_ios/ios/Classes/QuickActionsPlugin.swift

```diff
@@ -4,48 +4,45 @@
 
 import Flutter
 
-public final class QuickActionsPlugin: NSObject, FlutterPlugin {
+public final class QuickActionsPlugin: NSObject, FlutterPlugin, IOSQuickActionsApi {
 
   public static func register(with registrar: FlutterPluginRegistrar) {
-    let channel = FlutterMethodChannel(
-      name: "plugins.flutter.io/quick_actions_ios",
-      binaryMessenger: registrar.messenger())
-    let instance = QuickActionsPlugin(channel: channel)
-    registrar.addMethodCallDelegate(instance, channel: channel)
+    let messenger = registrar.messenger()
+    let instance = QuickActionsPlugin(messenger: messenger)
+    IOSQuickActionsApiSetup.setUp(binaryMessenger: messenger, api: instance)
     registrar.addApplicationDelegate(instance)
   }
 
-  private let channel: MethodChannel
   private let shortcutItemProvider: ShortcutItemProviding
-  private let shortcutItemParser: ShortcutItemParser
+  private let shortcutItemParser: ShortcutItemParser = DefaultShortcutItemParser()
+  private let flutterApi: IOSQuickActionsFlutterApiProtocol
   /// The type of the shortcut item selected when launching the app.
   private var launchingShortcutType: String? = nil
 
+  // This init is meant for unit testing only.
   init(
-    channel: MethodChannel,
-    shortcutItemProvider: ShortcutItemProviding = UIApplication.shared,
-    shortcutItemParser: ShortcutItemParser = DefaultShortcutItemParser()
+    flutterApi: IOSQuickActionsFlutterApiProtocol,
+    shortcutItemProvider: ShortcutItemProviding = UIApplication.shared
   ) {
-    self.channel = channel
+    self.flutterApi = flutterApi
     self.shortcutItemProvider = shortcutItemProvider
-    self.shortcutItemParser = shortcutItemParser
   }
 
-  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
-    switch call.method {
-    case "setShortcutItems":
-      // `arguments` must be an array of dictionaries
-      let items = call.arguments as! [[String: Any]]
-      shortcutItemProvider.shortcutItems = shortcutItemParser.parseShortcutItems(items)
-      result(nil)
-    case "clearShortcutItems":
-      shortcutItemProvider.shortcutItems = []
-      result(nil)
-    case "getLaunchAction":
-      result(nil)
-    case _:
-      result(FlutterMethodNotImplemented)
-    }
+  convenience init(
+    messenger: FlutterBinaryMessenger,
+    shortcutItemProvider: ShortcutItemProviding = UIApplication.shared
+  ) {
+    let shortcutItemProvider = shortcutItemProvider
+    let flutterApi = IOSQuickActionsFlutterApi(binaryMessenger: messenger)
+    self.init(flutterApi: flutterApi, shortcutItemProvider: shortcutItemProvider)
+  }
```

Comment:
Not sure if this is the best way to do this, but I was having a hard time figuring out a better way.

Reply:
Xcode insists this needs to be a convenience init